ocaml/libs/tracing/tracing.ml

```diff
@@ -71,11 +90,92 @@ module Spans = struct
 end
 
 module Tracer = struct
-  let start ~name ~parent : (Span.t option, exn) result =
-    let span = Span.start ~name ~parent () in
-    Spans.add_to_spans ~span ; Ok (Some span)
+  type t = {name: string; provider: provider_config_t ref}
+
+  let create ~name ~provider = {name; provider}
+
+  let no_op =
+    let provider =
+      ref
+        {
+          name_label= ""
+        ; tags= []
+        ; endpoints= []
+        ; filters= []
+        ; processors= []
+        ; enabled= false
+        }
+    in
+    {name= ""; provider}
+
+  let start ~tracer:t ~name ~parent : (Span.t option, exn) result =
+    let provider = !(t.provider) in
+    (* Do not start span if the TracerProvider is diabled*)
+    if not provider.enabled then
+      Ok None
```

Comment:
This creates a new 'Ok None' value everytime. Since we probably want to consider tracing disabled as the fastpath it might be useful to have a preallocated `let ok_none = Ok None`. This is what Lwt does to optimize common pattern like `Lwt.return ()` and `Lwt.return None` (it has a `Lwt.return_unit` and `Lwt.return_none`).

Reply:
@snwoods

Reply:
Resolved by 6f41e5ca3deac76ae8fb2b982df6fb4777f9f39e